great_expectations/data_context/store/inline_store_backend.py

```diff
@@ -100,6 +100,10 @@ def _set(self, key: tuple[str, ...], value: Any, **kwargs: dict) -> None:
 
         if resource_type is DataContextVariableSchema.ALL_VARIABLES:
             config_commented_map_from_yaml = yaml.load(value)
+            # NOTE: fluent datasources may be present under both the `fluent_datasources` & `datasources` key
```

Comment:
How do fluent datasources get added to the `datasources` key? Could we make this never happen?

Reply:
Because of the `self._datasource_store.set(key=None, value=datasource)` call that's needed to invoke the CloudAPI `DatasourceStore` webservice on a `CloudDataContext`

@cdkini are working on making this not a problem and then we can remove this but it may require some not insignificant changes to our `Stores`. But this unblocks the rest of the cloud work while we make those changes.

Reply:
Sounds good, thanks for the explanation.